File: result/translate/starchat/java2cpp/c++/LONGEST_REPEATING_SUBSEQUENCE.cpp
int f_gold ( string str ) {
    int n = str.length();
    vector<vector<int>> dp(n+1, vector<int>(n+1));
    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= n; j++) {
            if (i!= j && str[i-1] == str[j-1]) dp[i][j] = 1 + dp[i-1][j-1];
            else dp[i][j] = max(dp[i][j-1], dp[i-1][j]);
        }
    }
    return dp[n][n];
}

File: result/translate/transcoder/java2python/python/LONGEST_REPEATING_SUBSEQUENCE.py
def f_gold ( str ) :
    n = len ( str )
    dp = [ 0 ] * n + [ 0 ] * n + [ 0 ] * n
    for i in range ( 1 , n + 1 ) :
        for j in range ( 1 , n + 1 ) :
            if str [ i - 1 ] == str [ j - 1 ] and i != j :
                dp [ i ] [ j ] = 1 + dp [ i - 1 ] [ j - 1 ]
            else :
                dp [ i ] [ j ] = max ( dp [ i ] [ j - 1 ] , dp [ i - 1 ] [ j ] )
    return dp [ n ] [ n ]
